app/boards/arm/nice_nano/Kconfig

```diff
@@ -5,3 +5,9 @@ config BOARD_ENABLE_DCDC
     select SOC_DCDC_NRF52X
     default y
     depends on (BOARD_NICE_NANO || BOARD_NICE_NANO_V2)
+
+config BOARD_ENABLE_DCDC_HV
+    bool "High voltage DCDC converter"
+    select SOC_DCDC_NRF52X_HV
+    default y
+    depends on (BOARD_NICE_NANO_V2)
```

Comment:
Does this cause a Kconfig warning when building for v1? Another option would be `default y if ...` or `imply ...` under `BOARD_...` Kconfig.

Reply:
I don't see any warnings related to this in https://github.com/zmkfirmware/zmk/actions/runs/6512077486/job/17689044001?pr=1965

The goal is to only define this config for V2. V1 doesn't use high voltage mode, so there's no point in giving the option to enable this.

Reply:
Right, I wondered if it gives a warning like "you tried to enable `SOC_DCDC_NRF52X_HV` but its dependency `BOARD_NICE_NANO_V2` was not satisfied" because of `default y`, even if you didn't force it specifically. But that doesn't seem to be the case.